modules/thermalsensor_generic.c

```diff
@@ -0,0 +1,1451 @@
+ * the update notification before returning to the caller,
+ * but in order to minimize chances of deep recursion due
+ * to meta sensor handling, the notification is sent via
+ * idle callback.
+ *
+ * @param object   thermal object
+ *
+ * @return true if reading could be initiated, false otherwise
+ */
+static bool
+thermal_sensor_generic_read_sensor_cb(thermal_object_t *object)
+{
+    bool ack = false;
+
+    thermal_sensor_generic_t *self =
+        thermal_sensor_generic_from_object(object);
+
+    if( !thermal_sensor_generic_read_sensor(self) )
+        goto EXIT;
+
+#if 0
+    thermal_object_handle_update(object);
+#else
+    if( !self->sg_notify_id ) {
+        self->sg_notify_id =
+            g_idle_add(thermal_sensor_generic_sensor_notify_cb, object);
+    }
+#endif
+    ack = true;
+
+EXIT:
+    return ack;
+}
+
+/* ========================================================================= *
+ * CONFIGURATION_FILE
+ * ========================================================================= */
+
+/** Locate thermal object from list by name
+ *
+ * @param list  head of linked list
+ * @param name  name of the sensor to find
+ *
+ * @return thermal object, or NULL if not found
+ */
+static thermal_object_t *
+tsg_objects_get_object(GSList *list, const char *name)
+{
+    for( GSList *item = list; item; item = item->next ) {
+        thermal_object_t *object = item->data;
+
+        if( !object )
+            continue;
+
+        if( thermal_object_has_name(object, name) )
+            return object;
+    }
+
+    return 0;
+}
+
+/** Locate/add thermal object in list of sensors
+ *
+ * @param list  pointer to the head of linked list
+ * @param name  name of the sensor to find
+ *
+ * @return thermal object
+ */
+static thermal_object_t *
+tsg_objects_add_object(GSList **list, const char *name)
+{
+    thermal_object_t *object = tsg_objects_get_object(*list, name);
+
+    if( !object ) {
+        thermal_sensor_generic_t *sensor = thermal_sensor_generic_create(name);
+        object = thermal_object_create(&thermal_sensor_generic_vtab,
+                                       sensor);
+        *list = g_slist_prepend(*list, object);
+    }
+
+    return object;
+}
+
+/** Locate/add sensor object in list of sensors
+ *
+ * @param list  pointer to the head of linked list
+ * @param name  name of the sensor to find
+ *
+ * @return sensor object
+ */
+static thermal_sensor_generic_t *
+tsg_objects_add_sensor(GSList **list, const char *name)
+{
+    thermal_object_t *object = tsg_objects_add_object(list, name);
+    return thermal_sensor_generic_from_object(object);
+}
+
+/** Convert thermal limit name thermal status enum value
+ *
+ * @param key key string
+ *
+ * @return thermal status, or -1 on failure
+ */
+static THERMAL_STATUS
+tsg_objects_parse_level(const char *key)
+{
+    static const struct
+    {
+        const char *key;
+        int         val;
+    } lut[] = {
+        { "Low",     THERMAL_STATUS_LOW     },
```

Comment:
Named constants for these string literals maybe? Might make sense to have those in public headers?